.tekton/scripts/determine-if-e2e-execution-needed.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import json
+import urllib.error
+import urllib.request
+import subprocess
+
+# Pipelines currently covered by e2e tests
+pipelines_covered_by_e2e = ["docker-build", "docker-build-oci-ta", "docker-build-multi-platform-oci-ta", "fbc-builder"]
+
+# Task list which are covered by e2e tests, generated dynamically 
+tasks_covered_by_e2e = []
+
+# Otherthan tasks and pipelines, related files for which e2e tests needs to be executed
+files_covered_by_e2e = [".tekton/pull-request.yaml", ".tekton/tasks/e2e-test.yaml", ".tekton/scripts/determine-if-e2e-execution-needed.py"]
+
+def add_only_unique_task_names(task_list):
+    for task_name in task_list:
+        if task_name not in tasks_covered_by_e2e:
+            tasks_covered_by_e2e.append(task_name)
+    
+def get_tasks_covered_by_e2e():
+    for pipeline_name in pipelines_covered_by_e2e:
+        pipeline_path = f"pipelines/{pipeline_name}/{pipeline_name}.yaml"
+        # Get the task names from pipeline spec
+        result = subprocess.run([f"yq -e '.spec.tasks[].taskRef.name' {pipeline_path}"], shell=True, capture_output=True, text=True)
+        if result.stderr != "":
+            print(f"[ERROR] failed to get tasks inside spec.tasks: {result.stderr}")
+            sys.exit(1)
+        output = result.stdout
+        task_names = output.split()
+        add_only_unique_task_names(task_names)
+        # Get the task names from pipeline finally
+        result = subprocess.run([f"yq -e '.spec.finally[].taskRef.name' {pipeline_path}"], shell=True, capture_output=True, text=True)
+        if result.stderr != "":
+            print(f"[ERROR] failed to get tasks inside .spec.finally: {result.stderr}")
+            sys.exit(1)
+        output = result.stdout
+        task_names = output.split()
+        add_only_unique_task_names(task_names)
+
+def get_changed_files_from_pr(pull_number):
+    updated_files = []
+    token = os.environ["GITHUB_TOKEN"]
+    headers = {"Authorization": "Bearer " + token}
```

Comment:
To make this script runnable locally, could we make the GITHUB_TOKEN optional?

Reply:
To clarify, I think taking the list of changed files from the github API is OK. But it doesn't actually require authentication:

```
curl https://api.github.com/repos/konflux-ci/build-definitions/pulls/1903/files
```

^ works without the auth header. So my suggestion was just to not add the header if GITHUB_TOKEN is not defined.

But reworking it to use `git diff` is OK too

Reply:
@chmeliik  removed dependency on the GITHUB_TOKEN now. Please take a look again.